src/sys/socket/addr.rs

```diff
@@ -1006,8 +1006,8 @@ pub struct SockaddrIn(libc::sockaddr_in);
 impl SockaddrIn {
     /// Returns the IP address associated with this socket address, in native
     /// endian.
-    pub const fn ip(&self) -> libc::in_addr_t {
-        u32::from_be(self.0.sin_addr.s_addr)
+    pub fn ip(&self) -> net::Ipv4Addr {
```

Comment:
It seems that we can also retain the `const` here if we choose the following impl:

```rs
    pub const fn ip(&self) -> Ipv4Addr {
        let be_bytes = self.0.sin_addr.s_addr.to_be_bytes();
        let (a, b, c, d) = (be_bytes[0], be_bytes[1], be_bytes[2], be_bytes[3]);
        Ipv4Addr::new(a, b, c, d)
    }
```